Add language around default branch name (#48)
swift-server/sswg: process/incubation.md

* General
  * Has relevance to Swift on Server specifically
  * Publicly accessible source managed by an SCM such as github.com or similar

>     * Prefer to use `main` as the default branch name, in line with [Swift itself](https://forums.swift.org/t/moving-default-branch-to-main/38515) and all SSWG-owned repos

Reviewer: indent intentional?

Author: since it's related to SCM use I thought it made sense here

Reviewer 2: The phrasing of SSWG-owned sounds a bit weird to me, seems like a new term? 

I know what you mean and it's kind of right but the phrasing feels a bit off: Is it really that SSWG "owns" repos? Maybe we can use stronger language and only refer to the swift guidelines to emphasise the point?

I would not want to create the impression that in order to "be a valuable project in SSWG ecosystem I somehow have to let go of ownership" etc, which sometimes I feel it seems like to people, while we're just a collection of projects, owned by various people, who try to keep a same style and standard of quality etc.

```suggestion
    * Use `main` as the default branch name, in line with [Swift's guidelines](https://forums.swift.org/t/moving-default-branch-to-main/38515)
```

wdyt?

Author: sgtm to remove that wording, was mainly just trying to communicate that SSWG has/will do this for all of the repos under the `swift-server` org - but I think Swift's guidelines are enough to reference on their own.

there was some some discussion around whether we should _require_ this, and if so what impact that would have on existing repos in the incubation process, and I think the consensus was that it should be encouraged but not required.

another wording related note I thought of over the weekend; some people may already use something that is not `main` but is also not master as their main branch name (trunk, development, etc.) which accomplishes a lot of what we are trying to do here (we do lose the benefit of consistency across repos, but IMO that's less important)

Reviewer: imo recommending is a sufficiently strong guidelines in the context of incubation process. on the SSWG meeting we also discussed proposing a change to Swift's code of conduct that rejects such language more strongly beyond just branch name

orthogonally, we decided to change all repositories on swift-server org to use `main`

Author: propose we update to this to keep it a recommendation but also address @ktoso's concern
```suggestion
    * Prefer to use `main` as the default branch name, in line with [Swift's guidelines](https://forums.swift.org/t/moving-default-branch-to-main/38515)
```

Reviewer 2: Sounds good :) Thanks!
